### PR TITLE
update Django integration

### PR DESCRIPTION
**Pull request by tomkralidis**
# Overview
This PR provides the following updates on following #630:

- fixes Django dependencies to remove various CVEs
- adds source code and copyright headers
- removes files that should not be managed in version control
- updates documentation with instructions for running only (consistent with other web framework documentation), and thus moves that documentation into the bullet below
- moves sample application outside of Python installation into `examples/django`, with `README.md` per previous bullet

# Related Issue / Discussion
#630 
# Additional Information
None
# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by tomkralidis** [comment]
cc @lucadelu @krishnaglodha

File: .gitignore [hunk 1]
```diff
@@ -114,5 +114,3 @@ ENV/
 *.code-workspace
 
 .DS_Store
-
-pygeoapi/django_pygeoapi/sample_project/db.sqlite3
```
Comment (francbartoli) [inline .gitignore:118]: Is it now `examples/django/sample_project/db.sqlite3` to be ignored?

File: docs/source/running.rst [hunk 2]
```diff
@@ -181,6 +181,24 @@ is simple to run from the command, e.g:
 .. note::
    Uvicorn is as easy to install as ``pip install uvicorn``
 
+
+Django
+^^^^^^
+
+`Django`_ is a Python web framework that encourages rapid development and clean, pragmatic design.  Assuming
+a Django install/enabled application:
```
Comment (francbartoli) [inline docs/source/running.rst:189]: I would add this part coherently:

```console
HTTP request <--> Django (pygeoapi/django_app.py) <--> pygeoapi API (pygeoapi/api.py)

The Django server can be run as follows:

pygeoapi serve --django
```

File: docs/source/running.rst [hunk 3]
```diff
@@ -181,6 +181,24 @@ is simple to run from the command, e.g:
...
+
+.. code-block:: bash
+
+    python3 manage.py runserver
+
+In this section we create a sample Django project and use pygeoapi as an pluggable, embedded application.
```
Comment (francbartoli) [inline docs/source/running.rst:195]: I would rephrase with the common sentence

```text
To integrate pygeoapi as part of another Django project in a pluggable way the truly impatient developers can see `examples/django/sample_project/README.md` for a complete Django application.
As a result, your application will be available at http://localhost:5000/ and pygeoapi will be available at http://localhost:5000/oapi
```

File: docs/source/running.rst [hunk 4]
```diff
@@ -181,6 +181,24 @@ is simple to run from the command, e.g:
...
+
+.. note::
```
Comment (francbartoli) [inline docs/source/running.rst:197]: With the above comments we shouldn't need the note anymore

File: examples/django/sample_project/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+Update `urls.py` to run pygeoapi at e.g. `pga` path
+
+```python
+
+from django.contrib import admin
+from django.urls import path, include 
+from pygeoapi.django_pygeoapi import urls 
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('pga/', include(urls)) # added here
```
Comment (francbartoli) [inline examples/django/sample_project/README.md:68]: I would use `oapi` as replacement of `pga`, accordingly with the `running` documentation page

**Review by francbartoli — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (tomkralidis) [reply .gitignore:118]: Done

File: docs/source/running.rst (hunk 3 above)
Comment (tomkralidis) [reply docs/source/running.rst]: Done

File: docs/source/running.rst (hunk 4 above)
Comment (tomkralidis) [reply docs/source/running.rst]: note removed

File: examples/django/sample_project/README.md (hunk 5 above)
Comment (tomkralidis) [reply examples/django/sample_project/README.md]: done

File: docs/source/running.rst (hunk 2 above)
Comment (tomkralidis) [reply docs/source/running.rst:189]: I added the workflow to the `examples/django/sample_project/README.md` and added the rest to `docs/source/running.rst`.

**Comment by tomkralidis** [comment]
@francbartoli PR comments addressed, and ready for review.

**Review by francbartoli — APPROVED** [review]

**Comment by lucadelu** [comment]
I'd like to contribute to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.

**Comment by krishnaglodha** [comment]
I'd like to contribute Django downstream function to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.

**Comment by kalxas** [comment]
many thanks @lucadelu @krishnaglodha
